federation/pkg/federation-controller/sync/controller.go

```diff
@@ -320,126 +322,158 @@ func (s *FederationSyncController) reconcile(namespacedName types.NamespacedName
 		return statusNotSynced
 	}
 
-	key := namespacedName.String()
 	kind := s.adapter.Kind()
-	cachedObj, exist, err := s.store.GetByKey(key)
+	key := namespacedName.String()
+
+	obj, err := s.objFromCache(kind, key)
 	if err != nil {
-		glog.Errorf("failed to query main %s store for %v: %v", kind, key, err)
+		glog.Error(err)
 		return statusError
 	}
-
-	if !exist {
-		// Not federated, ignoring.
+	if obj == nil {
 		return statusAllOK
 	}
 
-	// Create a copy before modifying the resource to prevent racing
-	// with other readers.
-	copiedObj, err := api.Scheme.DeepCopy(cachedObj)
-	if err != nil {
-		glog.Errorf("error in retrieving %s from store: %v", kind, err)
-		return statusError
-	}
-	if !s.adapter.IsExpectedType(copiedObj) {
-		glog.Errorf("object is not the expected type: %v", copiedObj)
-		return statusError
-	}
-	obj := copiedObj.(pkgruntime.Object)
 	meta := s.adapter.ObjectMeta(obj)
-
 	if meta.DeletionTimestamp != nil {
-		if err := s.delete(obj, namespacedName); err != nil {
-			s.eventRecorder.Eventf(obj, api.EventTypeWarning, "DeleteFailed",
-				"%s delete failed: %v", strings.ToTitle(kind), err)
-			glog.Errorf("failed to delete %s %s: %v", kind, namespacedName, err)
+		err := s.delete(obj, kind, namespacedName)
+		if err != nil {
+			msg := "Failed to delete %s %q: %v"
+			args := []interface{}{kind, namespacedName, err}
+			glog.Errorf(msg, args...)
+			s.eventRecorder.Eventf(obj, api.EventTypeWarning, "DeleteFailed", msg, args...)
 			return statusError
 		}
 		return statusAllOK
 	}
 
-	glog.V(3).Infof("Ensuring delete object from underlying clusters finalizer for %s: %s",
-		kind, namespacedName)
-	// Add the required finalizers before creating the resource in underlying clusters.
+	glog.V(3).Infof("Ensuring finalizers for %s %q", kind, key)
```

Comment:
Nit: "Ensuring finalizers exist on %s %q

Or something else? This wording of "Ensure finalizers for" has always confused me a little bit.